test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -330,6 +349,10 @@ public void onRemoval(ShardId shardId, Accountable accountable) {
         namedWriteableRegistry = injector.getInstance(NamedWriteableRegistry.class);
     }
 
+    protected static void registerPlugin(Class<? extends Plugin> pluginClass) {
```

Comment:
instead of doing this, can we try to make `init()` a member method and see what's the impact? I mean if it's cheap why having the hassle to manage static deps?